### PR TITLE
feat: adding .editorconfig file to maintain code standards.

### LINKED ISSUE #143: Dev: Add editorconfig file to the project.
## Description
it would be good having editorconfig file in project to maintain the formatting style of whole project. 

## Acceptance Criteria
 file with accepted formatting style.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by nlok5923**
### Description

Added editorconfig file to maintain the code standards for the whole project.

Fixes #143 

### Type of Change:

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- Install editorconfig plugin in text editor
- plugin will directly detect .editorconfig file for code formatting rules

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by nlok5923** [comment]
@mtreacy002  please review this pr.

**Comment by mtreacy002** [comment]
@nlok5923 , can you please tell us how you test the code change (under the `How has this been tested` section)?

**Comment by nlok5923** [comment]
@mtreacy002 i had added how to test the code changes under ` how has this been tested section `

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[{package}.json]
```
Comment (mtreacy002) [inline .editorconfig]: @nlok5923 , why do we need to specify these again (index style and size) on [{package}.json] file type? won't line 3-5 above cover this type of file as well (with [*])?

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@nlok5923 . Please check my comment above.

File: .editorconfig (hunk 1 above)
Comment (nlok5923) [reply .editorconfig]: yes @mtreacy002  you were right will do that changes in the latest commit thanks.

**Comment by nlok5923** [comment]
@mtreacy002  please review again thanks

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+indent_style = space
```
Comment (Rahulm2310) [inline .editorconfig:4]: @nlok5923 Are you sure `space` would be better than `tab`  ?

Comment (nlok5923) [reply .editorconfig:4]: @Rahulm2310  i ventured through some of the React project and find space as good formatting standard.

**Review by mtreacy002 — DISMISSED** [review]
Well done, @nlok5923 and thank you for contributing to the project 🎉🎉🎉. 
I'm approving this now.

**Review by mtreacy002 — COMMENTED** [review]
@nlok5923 , you've done great on suggesting this .editorconfig. The code looks good to me. But before I approve the PR, wiill you be able to squash the commits from your end? Or do you want me to do that at my end instead?

**Comment by nlok5923** [comment]
@mtreacy002 thanks for approving the pull request I will squash commits from my end .

Thanks

**Comment by nlok5923** [comment]
@mtreacy002  i had squashed the commits now this pr is ready to merge 🎉

**Review by mtreacy002 — APPROVED** [review]
It LGTM now, @nlok5923 . Thank you for your contribution 🎉🎉🎉

**Pull request merged**